### PR TITLE
Release branch - 2.5.1

### PR DESCRIPTION
**Pull request by SteveDMurphy**
### Code Changes Included in Patch Release

* [x] [#2330](https://github.com/ethyca/fides/pull/2330)
* [x] [#2335](https://github.com/ethyca/fides/pull/2335)
* [x] [#2321](https://github.com/ethyca/fides/pull/2321)

### General

- [x] Quickstart verified working and up-to-date
- [x] Add any new tables/columns to the [database diagram](https://github.com/ethyca/fides/blob/5a485387d8af247ec6479e4115088cbbb8394d77/docs/fides/docs/development/update_erd_diagram.md) if they have not already been added
  - To check for the new tables/columns that have been added in the release, look for new migration files in the `src/fides/api/ctl/migrations` directory and inspect their contents
  - If found, look at the latest database diagram, and check whether it contains the tables/columns that were added in these migrations
  - If the diagram is not up to date, follow the steps in the link above to generate a new database diagram with the up-to-date schema, ensuring that all tables and their relationships can be clearly seen
  - Open up an issue on [fidesdocs](https://github.com/ethyca/fidesdocs/issues) to update the diagram on the new docs site
- [x] `nox -s test_env` works (verify the admin UI, privacy center, CLI and webserver)
- [x] `nox -s "build(sample)"` works on the release branch, creating the sample images (this is also prereq for `fides deploy up`)
- [x] `fides deploy up --no-pull` works using the images built in previous step (verify the admin UI, privacy center, CLI and webserver)

```
mkdir ~/fides-2-1-0-test
cd ~/fides-2-1-0-test
python3 -m venv venv
source venv/bin/activate
pip install git+https://github.com/ethyca/fides.git@<branch>
fides deploy up --no-pull
fides status
fides deploy down
rm -rf ~/fides-2-1-0-test
exit
```

Next, run the following checks against the environment you've spun up using `fides deploy up --no-pull`:

### API

- [x] Verify that the generated API docs are correct
- [x] Verify that the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CLI

Run these from within `nox -s dev -- shell`

- [x] Run a `fides push`
- [x] Run a `fides pull`
- [x] Run a `fides evaluate`
- [x] Generate a dataset with `fides generate dataset db --credentials-id app_postgres test.yml`
- [x] Scan a database with `fides scan dataset db --credentials-id app_postgres`

### Admin UI

- [x] Every navigation button works
- [x] DSR approval succeeds
- [x] DSR execution succeeds

### Privacy Center

- [x] Every navigation button works
- [x] DSR submission succeeds
- [x] Consent request submission succeeds

### Documentation

- [x] Verify that the CHANGELOG is formatted correctly and clean up verbiage where needed
- [x] Verify that the CHANGELOG is representative of the actual changes

## Post-Release Steps

- [x] Verify the ethyca-fides release is published to PyPi: <https://pypi.org/project/ethyca-fides/#history>
- [x] Verify the fides release is published to DockerHub: <https://hub.docker.com/r/ethyca/fides>
- [x] Verify the fides-privacy-center release is published to DockerHub: <https://hub.docker.com/r/ethyca/fides-privacy-center>
- [x] Verify the fides-sample-app release is published to DockerHub: <https://hub.docker.com/r/ethyca/fides-sample-app>
- [ ] Smoke test the PyPi & DockerHub releases with a clean `pip install ethyca-fides` and `fides deploy up`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,16 @@ The types of changes are:
 ## [Unreleased](https://github.com/ethyca/fides/compare/2.5.0...main)
```
Comment (seanpreston) [inline CHANGELOG.md:17]: This needs to say `2.5.1...main`

**Comment by SteveDMurphy** [comment]
@seanpreston - I am going to start running through the release steps. Since we aren't doing anything off `main`, I don't think we need to blast anything out until the release is shipped - good with that?

**Comment by SteveDMurphy** [comment]
One question about `nox -s text_env` - It looks like there are duplicate logs for `Loaded config from: /fides/src/fides/data/test_env/fides.test_env.toml` from the CLI but I thought we silenced that at some point recently? 🤔

**Comment by SteveDMurphy** [comment]
Minor question on if API docs version should be updated 
image

**Comment by SteveDMurphy** [comment]
Manually pushed the `latest` tags for privacy center and sample app again until #2301 is closed

**Comment by ThomasLaPiana** [comment]
@seanpreston @SteveDMurphy is this good to merge/close?

**Comment by ThomasLaPiana** [comment]
> Minor question on if API docs version should be updated image

oh nice catch...I'll make sure we inject the version into FastAPI

**Comment by ThomasLaPiana** [comment]
> One question about `nox -s text_env` - It looks like there are duplicate logs for `Loaded config from: /fides/src/fides/data/test_env/fides.test_env.toml` from the CLI but I thought we silenced that at some point recently? 🤔

we only silenced the "warning" message about no config found. but if there is one, we do write out where it's getting loaded from

**Comment by SteveDMurphy** [comment]
> > @seanpreston @SteveDMurphy is this good to merge/close?
> 
> Yep! Just waiting on an approval here still but would love to get it sorted before any changelog conflicts start rolling in - thanks @ThomasLaPiana

Thanks for the update on what was silenced! The second part here is the duplication of logs, I'll see if I can reproduce and open a bug around it tonight if that makes sense to you too 👍🏽

**Comment by ThomasLaPiana** [comment]
It looks good to me but I don't have full context here so I'll leave the real approval to @seanpreston

**Review by seanpreston — APPROVED** [review]

**Pull request merged**